samples/Basic_NFC/app/application.cpp

```diff
@@ -0,0 +1,90 @@
+#include <user_config.h>
+#include <SmingCore/SmingCore.h>
+#include "SerialReadingDelegateDemo.h"
+#include <AppSettings.h>
+#include "MFRC522.h"
+Timer procTimer;
+static Timer nfcScanTimer;
+SerialReadingDelegateDemo delegateDemoClass;
+int helloCounter = 0;
+
+void ICACHE_FLASH_ATTR scanNfc(byte scanner);
+//#define RST_PIN	15 // RST-PIN f�r RC522 - RFID - SPI - Modul GPIO15
+#define SS_PIN	4  	// D2
+//#define SS_PIN	15  // D8
+
+//MFRC522 mfrc522(SS_PIN, RST_PIN);	// Create MFRC522 instance
+MFRC522 mfrc522(SS_PIN, SS_PIN);	// Create MFRC522 instance
+byte ss_pin[]={4,15};
+
+void sayHello()
+{
+	for(int pinNdx=0;pinNdx<2;pinNdx++){
+		byte pin=ss_pin[pinNdx];
+		mfrc522.setControlPins(pin,pin);
+		mfrc522.PCD_Init();    // Init MFRC522
+		scanNfc(pin);
+	}
+
+
+}
+//----------------------------------
+// Helper routine to dump a byte array as hex values to Serial
+void dump_byte_array(byte *buffer, byte bufferSize) {
+  for (byte i = 0; i < bufferSize; i++) {
+    Serial.print(buffer[i] < 0x10 ? " 0" : " ");
+    Serial.print(buffer[i], HEX);
+  }
+}
+//---------------------------------
+void ICACHE_FLASH_ATTR
+scanNfc(byte scanner){
+static int cnt;
+	if ( ! mfrc522.PICC_IsNewCardPresent()) {
+		Serial.printf("Scanning nfc Scanner:%d \r\n", scanner);
```

Comment:
For consistency it will be better to use only debugf(...) for displaying debug information. Can you, please, replace `Serial.printf()` calls used to display debug information with `debugf()`?